private/buf/bufgen/internal/bufgenv2/config.go

```diff
@@ -22,34 +22,27 @@ import (
 	"github.com/bufbuild/buf/private/buf/buffetch"
 	"github.com/bufbuild/buf/private/buf/bufgen/internal"
 	"github.com/bufbuild/buf/private/buf/bufgen/internal/bufgenplugin"
-	"github.com/bufbuild/buf/private/bufpkg/bufimage"
+	"github.com/bufbuild/buf/private/bufpkg/bufimage/bufimagemodify/bufimagemodifyv2"
+	"github.com/bufbuild/buf/private/bufpkg/bufmodule/bufmoduleref"
 	"github.com/bufbuild/buf/private/pkg/normalpath"
 	"github.com/bufbuild/buf/private/pkg/storage"
 	"go.uber.org/zap"
 )
 
-// TODO: remove this
-func SilenceLinter() {
-	empty := ExternalConfigV2{}
-	_, _ = newManagedConfig(empty.Managed)
-	_ = validateExternalManagedConfigV2(empty.Managed)
-	_, _ = newDisabledFunc(ExternalManagedDisableConfigV2{})
-	_, _ = newOverrideFunc(ExternalManagedOverrideConfigV2{})
-	matchesModule(nil, "")
-	matchesPath(nil, "")
-	mergeDisabledFuncs(nil)
-	mergeOverrideFuncs(nil)
-	mergeFileOptionToOverrideFuncs(nil)
-	_ = applyManagement(nil, nil)
-}
-
-// DisableFunc decides whether a file option should be disabled for a file.
-type disabledFunc func(FileOption, bufimage.ImageFile) bool
+// disableFunc decides whether a file option should be disabled for a file.
+type disabledFunc func(fileOption, imageFileIdentity) bool
 
-// TODO: likely want something like *string or otherwise, see https://github.com/bufbuild/buf/issues/1949
 // overrideFunc is specific to a file option, and returns what thie file option
 // should be overridden to for this file.
-type overrideFunc func(bufimage.ImageFile) (string, error)
+type overrideFunc func(imageFileIdentity) bufimagemodifyv2.Override
+
+// imageFileIdentity is an image file that can be identified by a path and module identity.
+// There two (path and module) are the only information needed to decide whether to disable
+// or override a file option for a specific file. Using an interface to for easier testing.
+type imageFileIdentity interface {
```

Comment:
link to [discussion on a previous PR](https://github.com/bufbuild/buf/pull/2226#discussion_r1244148983)